Projects/DataSource/Sources/NetworkService/NetworkService.swift

```diff
@@ -11,17 +11,64 @@ import Shared
 final class NetworkService {
     static let shared = NetworkService()
     private let decoder = JSONDecoder()
+    private let plugins: [NetworkPlugin]
+    private let maxRetryCount = 1
 
-    private init() { }
+    private init() {
+        plugins = [
+            TokenInjectionPlugin(),
+            RefreshTokenPlugin(),
+            RetryPlugin()]
+    }
+
+    func request<T: Decodable>(
+        endpoint: Endpoint,
+        type: T.Type,
+        withPlugins: Bool = true
+    ) async throws -> T? {
+        var retryCount = 0
 
-    func request<T: Decodable>(endpoint: Endpoint, type: T.Type) async throws -> T? {
+        while true {
+            do {
+                return try await performRequest(endpoint: endpoint, type: type, withPlugins: withPlugins)
```

Comment:
```suggestion
                return try await performRequest(
                    endpoint: endpoint, 
                    type: type, 
                    withPlugins: withPlugins)
```

컨벤션에 의하면 3개부터 이르케 해야 합니당 ~~

Reply:
감사합니다~! 수정하겠습니다요